test/python/test_cutedsl_version_gate.py

```
@@ -0,0 +1,79 @@
+
+def test_decode_update_route_declines_by_version_before_importing_the_kernel(dsl_state, monkeypatch):
+    from cudnn.ops import _causal_conv1d_update as update
+
+    dsl_state(True, ("nvidia-cutlass-dsl", "4.6.2"))
+    monkeypatch.setattr(update, "_validate_semantic_contract", lambda *args: None)
+    monkeypatch.setattr(update, "_require_native_subset", lambda *args: None)
+    x = torch.zeros(2, 8, dtype=torch.bfloat16)
+    conv_state = torch.zeros(2, 8, 3, dtype=torch.bfloat16)
+    weight = torch.zeros(8, 4, dtype=torch.bfloat16)
+
+    with pytest.raises(NotImplementedError, match="found 4.6.2"):
```

Ruff (0.16.3)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)